packages/core/core/src/RequestGraph.js

```diff
@@ -150,6 +153,9 @@ export default class RequestGraph extends Graph<RequestGraphNode> {
   async transform(request: AssetRequest) {
     try {
       let start = Date.now();
+      if (!(await realpath(request.filePath)).includes('node_modules')) {
+        await this.runValidate(request);
```

Comment:
Do we actually need to await this here or could we run this in the background? For development mode, it might be nice to not block the build based on type checking.

Reply:
depends, if it throws errors it should probably be awaited as we'd otherwise have uncaught exception errors.

Although I think after the slack discussion it's safe to say we probably never wanna fail a build and just throw warnings.

Reply:
Maybe we could just queue the validations here and then run them after the build is complete?

Reply:
Not sure what the benefit of that would be unless we'd spawn a dedicated worker to run all typechecking. as that'll allow ts to keep a state and not reparse configs, recheck files and whatnot

Reply:
The benefit is faster builds in development. When changing code in development, we want to compile as quickly as possible and trigger HMR. TS can happen in the background after that. This means not using the workers for TS checking until after the build is complete.

Reply:
The idea of keeping state is interesting too. We could have a dedicated "validator" worker to run TS and other linters in maybe... I think that's how [fork-ts-checker-webpack-plugin](https://github.com/Realytics/fork-ts-checker-webpack-plugin) works, and fathy's plugin looks like it has its own worker too...

Reply:
Yes most plugins do it that way. I think it's the most performant way of doing this. Haven't experimented with running all workers with a TypeChecking state but I guess it wouldn't be much faster or even slower and just waste resources. I'll experiment with both once the TS PR is in and some SourceMap bugs are fixed.
This validator PR is not really high priority anyway and it was just a bit of an experiment to see if and how we'd support this.